VoodooI2CHID/VoodooI2CTouchscreenHIDEventDriver.cpp

```diff
@@ -249,10 +243,14 @@ void VoodooI2CTouchscreenHIDEventDriver::forwardReport(VoodooI2CMultitouchEvent
         // Send multitouch information to the multitouch interface
 
         if (event.contact_count >= 2) {
+            // Immediately cancel any outstanding click
+            timer_source->cancelTimeout();
+            drag_timer_source->cancelTimeout();
+
+            absolutetime_to_nanoseconds(timestamp, &last_multitouch_interaction);
+
             if (event.contact_count == 2 && start_scroll) {
                 scrollPosition(timestamp, event);
-            } else if (event.contact_count == 2 && !start_scroll) {
-                scheduleLift();
```

Comment:
I'm wondering if this might cause a problem. I removed it since lifts are now also clicks, but maybe it's required?

Reply:
It was! This was how the driver knew that a scroll ended. Without this, it didn't position the cursor between the two fingers when a scroll started. I introduced a new timer to handle this.